services/config.go

```diff
@@ -59,3 +61,26 @@ func (cfg Configuration) NewOptions() Options {
 	}
 	return opts
 }
+
+// ElectionConfiguration is for configuring election timeouts and TTLs
+type ElectionConfiguration struct {
+	LeaderTimeout *time.Duration `yaml:"leaderTimeout"`
+	ResignTimeout *time.Duration `yaml:"resignTimeout"`
+	TTLSeconds    *int           `yaml:"TTLSeconds"`
+}
+
+// NewOptions creates an ElectionOptions.
+func (cfg ElectionConfiguration) NewOptions() ElectionOptions {
+	opts := NewElectionOptions()
```

Comment:
I also prefer to order the setters in the same way as the fields in the config struct.